Examples/ObjectiveC/TextFormattingExample.m

```diff
@@ -0,0 +1,83 @@
+#import "TextFormattingExample.h"
+@import Mapbox;
+
+NSString *const MBXExampleFormattingExpression = @"TextFormattingExample";
+
+@interface TextFormattingExample ()
+
+@property (nonatomic) MGLMapView *mapView;
+
+@end
+
+@implementation TextFormattingExample
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.mapView = [[MGLMapView alloc] initWithFrame: CGRectMake(0, 0, self.view.bounds.size.width, self.view.bounds.size.height)];
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    
+    self.mapView.styleURL = [MGLStyle streetsStyleURL];
+    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(36.374782, -119.620859)
+                            zoomLevel:6
+                             animated:NO];
+    [self.view addSubview:self.mapView];
+    
+    UISegmentedControl *styleToggle =[[UISegmentedControl alloc] initWithItems:@[@"Expression", @"JSON"]];
+    styleToggle.translatesAutoresizingMaskIntoConstraints = NO;
+    styleToggle.tintColor = [UIColor colorWithRed:0.976 green:0.843 blue:0.831 alpha:1];
+    styleToggle.backgroundColor = [UIColor colorWithRed:0.973 green:0.329 blue:0.294 alpha:1];
+    styleToggle.layer.cornerRadius = 4;
+    styleToggle.clipsToBounds = YES;
+    [self.view insertSubview:styleToggle aboveSubview:self.mapView];
+    [styleToggle addTarget:self action:@selector(formatText:) forControlEvents:UIControlEventValueChanged];
+    
+    NSMutableArray *constraints = [NSMutableArray array];
+    
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:styleToggle attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:self.mapView attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:1.0]];
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:styleToggle attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self.mapView.logoView attribute:NSLayoutAttributeTop multiplier:1 constant:-20]];
+    
+    [self.view addConstraints:constraints];
+}
+
+// Format label's text based on the selected index of the UISegmentedControl
+- (void)formatText:(UISegmentedControl *)sender {
+    MGLSymbolStyleLayer *stateLayer = (MGLSymbolStyleLayer *)[self.mapView.style layerWithIdentifier:@"state-label"];
```

Comment:
I recommend also mentioning that the above layers only exist in styles that use the `mapbox-streets-v8` source.